test/unit/cmap/auth/mongodb_oidc/callback_lock_cache.test.ts

```diff
@@ -0,0 +1,144 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import { setTimeout } from 'timers';
+
+import {
+  CallbackLockCache,
+  Connection,
+  MongoCredentials,
+  MongoInvalidArgumentError
+} from '../../../../mongodb';
+
+describe('CallbackLockCache', function () {
+  describe('#getCallbacks', function () {
+    const connection = sinon.createStubInstance(Connection);
+    connection.address = 'localhost:27017';
+
+    context('when a request callback does not exist', function () {
+      const credentials = new MongoCredentials({
+        username: 'test_user',
+        password: 'pwd',
+        source: '$external',
+        mechanismProperties: {}
+      });
+      const cache = new CallbackLockCache();
+
+      it('raises an error', function () {
+        try {
+          cache.getCallbacks(connection, credentials);
+        } catch (error) {
+          expect(error).to.be.instanceOf(MongoInvalidArgumentError);
+          expect(error.message).to.include(
+            'Auth mechanism property REQUEST_TOKEN_CALLBACK is required'
+          );
+        }
+      });
+    });
+
+    context('when no entry exists in the cache', function () {
+      context('when a refresh callback exists', function () {
+        let requestCount = 0;
+        let refreshCount = 0;
+
+        const request = async () => {
+          requestCount++;
+          if (requestCount > 1) {
+            throw new Error('Cannot execute request simultaneously.');
+          }
+          await new Promise(resolve => setTimeout(() => resolve(), 1000));
```

Comment:
We have a `sleep` util, in `test/tools/utils.ts`

Reply:
`sleep` didn't exist so I created it in utils, and changed all these to use it.

Reply:
Oh I missed that, thanks. I've updated to use the test tools sleep now.